netket/experimental/operator/_fermions_2nd.py

```diff
@@ -0,0 +1,769 @@
+        This class can be initialized in the following form: ``FermionOperator2nd(hilbert, terms, weights ...)``.
+        The term contains pairs of `(idx, dagger)`, where `idx ∈ range(hilbert.size)` (it identifies an orbital) and `dagger` is a True/False flag determining if the operator is a creation or destruction operator.
+        A term of the form :math:`\\hat{a}_1^\\dagger \\hat{a}_2` would take the form ((1,1), (2,0)), where (1,1) represents :math:`\\hat{a}_1^\\dagger` and (2,0) represents :math:`\\hat{a}_2`.
+        To split up per spin, use the creation and annihilation operators to build the operator.
+
+        Args:
+            hilbert (required): hilbert of the resulting FermionOperator2nd object
+            terms (list(str) or list(list(list(int)))): single term operators (see example below)
+            weights (list(union(float,complex))): corresponding coefficients of the single term operators
+            constant (float, complex): constant contribution (corresponding to the identity operator * constant)
+
+        Returns:
+            A FermionOperator2nd object.
+
+        Example:
+            Constructs a new ``FermionOperator2nd`` operator (0.5-0.5j)*(a_0^dagger a_1) + (0.5+0.5j)*(a_2^dagger a_1)  with the construction scheme.
+            >>> import netket.experimental as nkx
+            >>> terms, weights = (((0,1),(1,0)),((2,1),(1,0))), (0.5-0.5j,0.5+0.5j)
+            >>> hi = nkx.hilbert.SpinOrbitalFermions(3)
+            >>> op = nkx.operator.FermionOperator2nd(hi, terms, weights)
+            >>> op
+            FermionOperator2nd(hilbert=SpinOrbitalFermions(n_orbitals=3), n_operators=2)
+            >>> terms = ("0^ 1", "2^ 1")
+            >>> op = nkx.operator.FermionOperator2nd(hi, terms, weights)
+            >>> op
+            FermionOperator2nd(hilbert=SpinOrbitalFermions(n_orbitals=3), n_operators=2)
+            >>> op.hilbert
+            SpinOrbitalFermions(n_orbitals=3)
+            >>> op.hilbert.size
+            3
+        """
+        super().__init__(hilbert)
+        self._dtype = dtype
+
+        # bring terms, weights into consistent form
+        _operators, _constant = _canonicalize_input(terms, weights, constant, dtype)
+        # we keep the input, in order to be able to add terms later
+        self._operators = _operators
+        self._constant = _constant
+
+        self._initialized = False
+        self._is_hermitian = None  # set when requested
+
+    def _setup(self, force: bool = False):
+        """Analyze the operator strings and precompute arrays for get_conn inference"""
+        if force or not self._initialized:
+
+            # following lists will be used to compute matrix elements
+            # they are filled in _add_term
+            out = _pack_internals(self._operators, self._dtype)
+            self._orb_idxs, self._daggers, self._weights, self._term_ends = out
+
+            self._initialized = True
+
```

Comment:
the name maybe is not the best, but we need this.
And everytime we modify inplace we need to call this.

```suggestion

    def _reset_caches(self):
        """
        Cleans the internal caches built on the operator.
        """
        self._initialized = False
        self._is_hermitian = None
```